searchlib/src/vespa/searchlib/features/queryterm.cpp

```diff
@@ -3,45 +3,66 @@
 #include "queryterm.h"
 #include "utils.h"
 
-using namespace search::fef;
 using search::feature_t;
+using namespace search::fef;
 
 namespace search::features {
 
-QueryTerm::QueryTerm() :
-    _termData(nullptr),
-    _handle(IllegalHandle),
-    _significance(0),
-    _connectedness(0)
+QueryTerm
+QueryTermFactory::create(const IQueryEnvironment & env, uint32_t termIdx, bool lookupConnectedness)
 {
+    const ITermData *termData = env.getTerm(termIdx);
+    feature_t fallback = util::getSignificance(*termData);
+    feature_t significance = features::util::lookupSignificance(env, termIdx, fallback);
+    feature_t connectedness = 0;
+    if (lookupConnectedness) {
+        connectedness = util::lookupConnectedness(env, termIdx);
+    }
+    return QueryTerm(termData, significance, connectedness);
 }
 
-QueryTerm::QueryTerm(const ITermData * td, feature_t sig, feature_t con) :
-    _termData(td),
-    _handle(IllegalHandle),
-    _significance(sig),
-    _connectedness(con)
+QueryTermHelper::QueryTermHelper(const IQueryEnvironment &env)
+        : _fallBack(),
+          _queryTerms(lookupQueryTerms(env))
 {
+    if (_queryTerms == nullptr) {
+        _fallBack = createQueryTermvector(env);
+        _queryTerms = & _fallBack;
+    }
 }
 
-QueryTerm
-QueryTermFactory::create(const IQueryEnvironment & env,
-                         uint32_t termIdx,
-                         bool lookupSignificance,
-                         bool lookupConnectedness)
+namespace {
+
+using QueryTermVectorWrapper = AnyWrapper<QueryTermVector>;
+const vespalib::string QUERY_TERMS_KEY("queryvectorhelper.queryterms");
+
+}
+const QueryTermVector &
+QueryTermHelper::lookupAndStoreQueryTerms(const IQueryEnvironment &env, IObjectStore & store)
 {
-    const ITermData *termData = env.getTerm(termIdx);
-    feature_t significance = 0;
-    if (lookupSignificance) {
-        feature_t fallback = util::getSignificance(*termData);
-        significance = util::lookupSignificance(env, termIdx, fallback);
-    }
-    feature_t connectedness = 0;
-    if (lookupConnectedness) {
-        connectedness = search::features::util::lookupConnectedness(env, termIdx);
+    const Anything * obj = store.get(QUERY_TERMS_KEY);
+    if (obj == nullptr) {
+        store.add(QUERY_TERMS_KEY, std::make_unique<QueryTermVectorWrapper>(createQueryTermvector(env)));
+        obj = store.get(QUERY_TERMS_KEY);
     }
-    return QueryTerm(termData, significance, connectedness);
+    return static_cast<const QueryTermVectorWrapper *>(obj)->getValue();
```

Comment:
consider adding a static convenience function to the AnyWrapper template that casts the Anything pointer to a const T reference.

Reply:
Fixed